battery_monitor/src/battery_monitor.cpp

```diff
@@ -91,98 +91,137 @@ BatteryMonitor::BatteryMonitor(ros::NodeHandle &nodeHandle)
-      });
+      }
+);
   diagnosticsUpdater.add(batteryVoltageCheck);
 
   batteryTemperatureCheck = diagnostic_tools::create_health_check<BatteryInfo>(
-      "Battery Voltage check", [=](const BatteryInfo &info) -> diagnostic_tools::Diagnostic {
+      "Battery Voltage check",
+      [maxBatteryTemperature](const BatteryInfo &info) -> diagnostic_tools::Diagnostic
+      {
         using diagnostic_tools::Diagnostic;
         Diagnostic diagnostic{Diagnostic::OK};
-        if (info.systemThermocouple1 > maxBatteryTemperature) {
+        if (info.systemThermocouple1 > maxBatteryTemperature)
+        {
           diagnostic.status(Diagnostic::WARN)
               .description("Packs temperature - NOT OK (%f degC > %f degC)",
                            info.systemThermocouple1, maxBatteryTemperature)
               .code(health_monitor::ReportFault::BATTERY_TEMPERATURE_THRESHOLD_REACHED);
-        } else {
+        }
+        else
+        {
           diagnostic.description("Packs temperature - OK (%f degC)", info.systemThermocouple1);
         }
         return diagnostic;
-      });
+      }
+);
 
   diagnosticsUpdater.add(batteryTemperatureCheck);
 
   publisher_reportLeakDetected =
       diagnostic_tools::create_publisher<battery_monitor::ReportLeakDetected>(
           nodeHandle, "/battery_monitor/report_leak_detected", 1);
 
+  diagnostic_tools::PeriodicMessageStatusParams paramsBatteryLeakCheckPeriod{};
+  paramsBatteryLeakCheckPeriod.min_acceptable_period(minBatteryLeakCheckPeriod);
+  paramsBatteryLeakCheckPeriod.max_acceptable_period(maxBatteryLeakCheckPeriod);
+  diagnostic_tools::Diagnostic diagnosticBateryInfoLeakStale
+  {
+    diagnostic_tools::Diagnostic::WARN,
+    health_monitor::ReportFault::BATTERY_INFO_STALE
+  };
+  paramsBatteryLeakCheckPeriod.abnormal_diagnostic(diagnosticBateryInfoLeakStale);
+
   diagnosticsUpdater.add(
       publisher_reportLeakDetected.add_check<diagnostic_tools::PeriodicMessageStatus>(
-          "rate check",
-          diagnostic_tools::PeriodicMessageStatusParams{}
-              .min_acceptable_period(minBatteryLeakCheckPeriod)
-              .max_acceptable_period(maxBatteryLeakCheckPeriod)
-              .abnormal_diagnostic({diagnostic_tools::Diagnostic::WARN,
-                                    health_monitor::ReportFault::BATTERY_INFO_STALE})));
+          "rate check", paramsBatteryLeakCheckPeriod));
 
   batteryLeakCheck = diagnostic_tools::create_health_check<bool>(
-      "Battery leak check", [=](bool leak_detected) -> diagnostic_tools::Diagnostic {
+      "Battery leak check", [](bool leak_detected) -> diagnostic_tools::Diagnostic
+      {
         using diagnostic_tools::Diagnostic;
-        if (leak_detected) {
-          return {Diagnostic::WARN, "Battery leak detected",
+        if (leak_detected)
+        {
+          return
+          {
+                  Diagnostic::WARN, "Battery leak detected",
                   health_monitor::ReportFault::BATTERY_LEAK_DETECTED};
```

Comment:
@patapufate nit: consider adding a `using` statement for `health_monitor::ReportFault::BATTERY_LEAK_DETECTED`, so that this `return` fits in one line. Formatting here is awkward.

Reply:
Fixed

Reply:
Changed in all the constant defined in health_monitor::ReportFault